src/test/java/hudson/plugins/mercurial/MercurialSampleRepoRule.java

```diff
@@ -48,4 +51,20 @@ public void notifyCommit(JenkinsRule r) throws Exception {
         r.waitUntilNoActivity();
     }
 
+    public void registerHook(JenkinsRule r) throws Exception {
+        File hgDir = new File(sampleRepo, ".hg");
+        File hook = new File(hgDir, "hook.py");
+        FileUtils.write(hook, "import requests\n"
```

Comment:
Fails for me:

```
…
=== Starting modernHook(hudson.plugins.mercurial.PipelineTest)
Jan 04, 2017 10:55:58 AM hudson.plugins.mercurial.PipelineTest modernHook
INFO: $ hg version
Mercurial Distributed SCM (version 3.9.1)
(see https://mercurial-scm.org for more information)
Copyright (C) 2005-2016 Matt Mackall and others
This is free software; see the source for copying conditions. There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
adding Jenkinsfile
…
[p/default #1] Finished: SUCCESS
error: commit.jenkins hook raised an exception: No module named requests
(run with --traceback for stack trace)
---%<--- job/p/indexing/ EVENTS
…
java.io.FileNotFoundException: /tmp/jenkinsTests.tmp/jenkins1189541614147456129test/jobs/p/indexing/events.log (No such file or directory)
	at java.io.RandomAccessFile.open0(Native Method)
	at java.io.RandomAccessFile.open(RandomAccessFile.java:316)
	at java.io.RandomAccessFile.<init>(RandomAccessFile.java:243)
	at org.kohsuke.stapler.framework.io.LargeText$FileSession.<init>(LargeText.java:397)
	at org.kohsuke.stapler.framework.io.LargeText$2.open(LargeText.java:120)
	at org.kohsuke.stapler.framework.io.LargeText.writeLogTo(LargeText.java:210)
	at hudson.console.AnnotatedLargeText.writeLogTo(AnnotatedLargeText.java:154)
	at hudson.plugins.mercurial.PipelineTest.showEvents(PipelineTest.java:279)
	at hudson.plugins.mercurial.PipelineTest.modernHook(PipelineTest.java:231)
	at …
```

I tried

    sudo apt-get install python-requests

But now it just fails with:

```
error: commit.jenkins hook raised an exception: No module named urllib3.exceptions
```

despite having the `python-urllib3` package installed, providing `/usr/lib/python2.7/dist-packages/urllib3/exceptions.py`.

I really need to port the Docker fixture support from `acceptance-test-harness` so it can be used from functional tests, because this is getting out of hand.